native/src/main/java/io/ballerina/stdlib/websocket/actions/websocketconnector/Close.java

```diff
@@ -55,6 +57,9 @@ public static Object externClose(Environment env, BObject wsConnection, long sta
             List<BError> errors = new ArrayList<>(1);
             ChannelFuture closeFuture = initiateConnectionClosure(errors, (int) statusCode, reason.getValue(),
                     connectionInfo, countDownLatch);
+            if (connectionInfo.getWebSocketEndpoint().getType().getName().equals(SYNC_CLIENT)) {
+                connectionInfo.getWebSocketConnection().readNextFrame();
+            }
             waitForTimeout(errors, (int) timeoutInSecs.floatValue(), countDownLatch, connectionInfo);
```

Comment:
I think it would be good if we have the above code lines after the `closefuture`.